src/main/java/org/reactome/addlinks/dataretrieval/pharos/PharosTargetsDataRetriever.java

```diff
@@ -19,16 +19,39 @@
 			+ "    }"
 			+ "}\" }";
 
+	// Data from Pharos should look like this:
+	//	{
+	//		  "data": {
+	//		    "targets": {
+	//		      "count": 20412,
+	//		      "targets": [
+	//		        {
+	//		          "uniprot": "A8MVA2"
+	//		        },
+	//		        {
+	//		          "uniprot": "A8MX34"
+	//		        },
+	//		        ...
+
 
 	/**
 	 * Implements the logic to process JSON from Pharos.
 	 */
-	protected int processJSONArray(FileWriter fw, JSONObject jsonObj) throws IOException
+	protected int processJSONArray(FileWriter writer, JSONObject jsonObj) throws IOException
 	{
 		JSONArray targets = (((JSONObject)((JSONObject) jsonObj.get("data")).get("targets")).getJSONArray("targets"));
 		for (int i = 0; i < targets.length(); i++)
 		{
-			fw.write(((JSONObject)targets.get(i)).get("uniprot") + "\n");
+			if (((JSONObject)targets.get(i)).has("uniprot"))
+			{
+				writer.write(((JSONObject)targets.get(i)).get("uniprot") + "\n");
+			}
+			else
+			{
+				logger.error("The \"uniprot\" key was not present, but it should be (in fact, it should be the ONLY key at this depth in the document). Check your query!");
+				// Something's wrong with the data, so gracefully exit the loop.
+				i = targets.length() + 1;
```

Comment:
Could this be a `break;` instead?  I didn't understand immediately on reading that the loop is exiting by modifying the value of i to be one more than `targets.length()`.

Reply:
I don't think this would be an appropriate use of `break`.

Reply:
Is there a reason why?  `break;` is meant to stop a loop (or switch statement).  I think it is important because modifying the counter within the for loop usually isn't done outside of the for statement itself.

Reply:
Honestly, I have a hard time coming up with _appropriate_ uses of `break`.

I think that using `break` is a bad practice in general. There's almost always a way to clearly define the exit conditions from a loop. `break` leads to hidden traps and bad habits. I've seen code where break and continue were used in horrible ways to make code that was almost completely incomprehensible.

I don't understand what's wrong with explicitly violating the loop invariant to exit the loop early.

Reply:
Well... switch-case statements use `break`, but that's the only place I use them, and that's because they are designed that way.

Reply:
I am just very uncomfortable with modifying the loop condition in a for loop.  I've never seen it done.  I really do think this is an appropriate use for break, here.

If you'd prefer not to use break, I think this would be better as a while loop so that you can include the counter and the condition of error to stop.

Reply:
Modifying a variable to break a for-loop's condition is not common, but I have seen it done and I still think it's more appropriate than a `break`. I'll probably rewrite this as a while-loop since I think that's the only thing we'll agree on.

Reply:
I've changed the way this possible error gets handled. Please have a look.

Reply:
Looks good!  Thanks for making the change!